include/ttmlir/Target/Utils/MLIRToFlatbuffer.h

```diff
@@ -15,6 +16,29 @@
 #include "ttmlir/Utils.h"
 
 namespace mlir::tt {
+struct GoldenTensor {
+  std::string name;
+  std::vector<int64_t> shape;
+  std::vector<int64_t> strides;
+  ::tt::target::DataType dtype;
+  std::uint8_t *data;
+
+  GoldenTensor(std::string name, std::vector<int64_t> shape,
+               std::vector<int64_t> strides, ::tt::target::DataType dtype,
+               std::uint8_t *data)
+      : name(name), shape(shape), strides(strides), dtype(dtype), data(data) {}
+
+  std::vector<std::uint8_t> convertDataToVector() {
+    int totalDataSize = std::accumulate(this->shape.begin(), this->shape.end(),
+                                        1, std::multiplies<int64_t>());
+
+    std::vector<std::uint8_t> dataVec(totalDataSize * sizeof(float));
+    std::memcpy(dataVec.data(), this->data, totalDataSize * sizeof(float));
```

Comment:
Nitpick, feel free to ignore: move `* sizeof(float)` above `std::accumulate() * sizeof(float)` to avoid doing the multiplication two times (not that it speeds up anything, its just a bit cleaner) and leave comment saying all tensor elements are interpreted as floats.

Reply:
fixed.